### DISCUSSION TITLE
Add `remove` and `removeAll` to the plugins manager

### DISCUSSION THREAD

**Discussion by mamal72** [GrapesJS/grapesjs#4066] [Ideas]
I have a plugin that I'm loading inside another function. That function gets called in different places with different args and the plugin needs to be added again to respect that newly passed arg to my loader function.

GrapesJS does not call the `add` function in plugins manager if it was added once before. I'm proposing 2 new functions here to add this to the GrapesJS plugins manager.

- `remove: (id: string) => void`: To remove a specific plugin from the plugins object.
- `removeAll: () => void`: To remove all the plugins. This basically can be used as a cleanup method in more complex cases.

Right now, I'm doing this hack to reload my plugin with a new value:

```ts

// It won't work
const loadMyPlugin = (colors: string[]) => {
  Grapesjs.plugins.add('myPlugin', (editor, options) => {
    // It will be called only once and it won't get called if I call `loadMyPlugin` multiple times. So the `colors` value won't be updated from the outside world. 
  })
}

// Here's my hack to make it work
const loadMyPlugin = (colors: string[]) => {
  const plugins = Grapesjs.plugins.getAll()
  plugins['myPlugin'] = (editor, options) => {
    // It works as expected and it will replace the old plugin with this new one. So the `colors` value will be the updated new value.
  }
}

// Here's my suggestion for this case
const loadMyPlugin = (colors: string[]) => {
  Grapesjs.plugins.remove('myPlugin') // This new `remove` method will remove that plugin from the plugins manager if it does exist.
  Grapesjs.plugins.add('myPlugin', (editor, options) => {
    // It will be called only once and it won't get called if I call `loadMyPlugin` multiple times. So the `colors` value won't be updated from the outside world. 
  })
}
```

@artf Please let me know your thoughts on this. I'm gonna send a PR for this.

**Comment by mamal72** [comment]
Here's the related PR: #4067.

**Comment by artf** [comment]
I'm not sure removing simply the plugin from the object makes sense. It would be quite a more complex task than that (think about removing all added blocks, components, styles, etc.) if we really want to implement it.

In your case, I think you simply not doing the plugin in the proper way. Try to rethink it, you don't need to remove it, you might probably need a simple [stateful command](https://grapesjs.com/docs/modules/Commands.html#stateful-commands) which actives stuff on `run` and disables them on `stop`.
